### DISCUSSION TITLE
How can I get download url by Java Client?

### DISCUSSION THREAD

**Discussion by MerlinFenrir** [minio/minio#14616] [Q&A] [answered]
Hello, now I can download my file by getObject() api, but how can I get a download url instead of input stream? It's like getting the download address at the page end.

I run a standalone MinIO server on Linux, and Interact with a java springboot project clients.

**Comment by harshavardhana** [comment, marked as answer]
You know the URL, You know the bucket, you know the object - how hard it is to construct a URL? @MerlinFenrir

**Reply by MerlinFenrir** [reply]
but I can't get a special download url from Java Client API Reference, I just want get file from minio server but not java client

**Reply by reda-alaoui** [reply]
> how hard it is to construct a URL?

As hard as creating a 393 lines `buildUrl` method in `S3Base`.
